### PR TITLE
Propagate trace and parent span ids

### PR DESCRIPTION
**Pull request by nadiaciobanu**
Now that the `Span` class contains a `SpanContext` instance, we can propagate trace and parent span ids using `SpanContext`. When the tracer creates a new `Span`, get the `SpanContext` of the current span from the runtime context and pass it to the `Span` constructor. In the `Span` constructor, create a new `SpanContext` and use the parent span context to set the trace and parent span id.

I exported to GCP using the OTLP example (`example/otlp`) and verified that the trace and parent span ids propagate properly:
Screenshot 2020-08-28 at 11 14 00 AM

### CODE REVIEW

File: sdk/src/trace/span.cc [hunk 1]
```diff
@@ -77,16 +78,21 @@ Span::Span(std::shared_ptr<Tracer> &&tracer,
...
   else
   {
-    recordable_->SetIds(GenerateRandomTraceId(), GenerateRandomSpanId(), trace_api::SpanId());
+    trace_id = GenerateRandomTraceId();
+    recordable_->SetIds(trace_id, GenerateRandomSpanId(), trace_api::SpanId());
   }
-  // TODO: Create and populate SpanContext for this span when SpanContext is fully implemented
+
+  span_context_ = std::unique_ptr<trace_api::SpanContext>(
+      new trace_api::SpanContext(trace_id, GenerateRandomSpanId(), trace_api::TraceFlags(), false));
```
Comment (g-easy) [inline sdk/src/trace/span.cc:95]: Why is this a different[ly random] span ID from the one on L86 or L91?

File: sdk/test/trace/tracer_test.cc [hunk 2]
```diff
@@ -113,7 +113,7 @@ TEST(Tracer, ToMockSpanExporter)
   ASSERT_EQ("span 2", spans_received->at(0)->GetName());
   EXPECT_TRUE(spans_received->at(0)->GetTraceId().IsValid());
   EXPECT_TRUE(spans_received->at(0)->GetSpanId().IsValid());
-  EXPECT_FALSE(spans_received->at(0)->GetParentSpanId().IsValid());
+  EXPECT_TRUE(spans_received->at(0)->GetParentSpanId().IsValid());
```
Comment (g-easy) [inline sdk/test/trace/tracer_test.cc:116]: Do we also check the values of trace_id and parent_span_id match?

File: sdk/src/trace/span.cc (hunk 1 above)
Comment (nadiaciobanu) [reply sdk/src/trace/span.cc:95]: It shouldn't have been. Fixed now.

File: sdk/test/trace/tracer_test.cc (hunk 2 above)
Comment (nadiaciobanu) [reply sdk/test/trace/tracer_test.cc:116]: Added tests for this.

**Comment by nadiaciobanu** [comment]
@reyang Ready to merge :)

**Pull request merged**